src/tike/operators/propagation.py

```diff
@@ -6,11 +6,41 @@
 
 
 class Propagation(Operator):
-    """A base class for Fourier-based free-space propagation."""
+    """A base class for Fourier-based free-space propagation.
+
+    Take an (..., N, N) array and pads the last two dimensions with zeros until
+    it is size (..., M, M). Then apply the Fourier transform to the last two
+    dimensions.
+
+    Attributes
+    ----------
+    probe_shape : int
+        The pixel width and height of the nearplane waves.
+    detector_shape : int
+        The pixel width and height of the farplane waves.
+    nwaves : int
+        The number of waves to propagate from the nearplane to farplane.
+    model : string
+        The type of noise model to use for the cost functions.
+    cost : (data-like, farplane-like) -> float
+        The function to be minimized when solving a problem.
+    grad : (data-like, farplane-like) -> farplane-like
+        The gradient of cost.
```

Comment:
gradient of cost w.r.t. what?

Reply:
With respect to the farplane. data is a constant, so grad is a function of one variable.